heareval/tasks/nsynth_pitch.py

```diff
@@ -0,0 +1,225 @@
+
+        # Filter out pitches that are not within the range of a standard piano
+        metadata = metadata[metadata["pitch"] >= 21]
+        metadata = metadata[metadata["pitch"] <= 108]
+
+        metadata = metadata.assign(label=lambda df: df["pitch"])
+        metadata = metadata.assign(
+            relpath=lambda df: df["note_str"].apply(
+                partial(self.get_rel_path, split_path)
+            )
+        )
+        metadata = metadata.assign(
+            slug=lambda df: df["note_str"].apply(self.slugify_file_name)
+        )
+        metadata = metadata.assign(partition=lambda df: split)
+        metadata = metadata.assign(
+            filename_hash=lambda df: df["slug"].apply(luigi_util.filename_to_int_hash)
+        )
+
+        return metadata[luigi_util.PROCESSMETADATACOLS]
+
+    def run(self):
+
+        # Get metadata for each of the data splits
+        process_metadata = pd.concat(
+            [self.get_split_metadata(split) for split in self.requires()]
+        )
+
+        process_metadata.to_csv(
+            os.path.join(self.workdir, self.outfile),
+            columns=luigi_util.PROCESSMETADATACOLS,
+            header=False,
+            index=False,
+        )
+
+        self.mark_complete()
+
+
+class SubsamplePartition(luigi_util.SubsamplePartition):
+    """
+    A subsampler that acts on a specific partition.
+    All instances of this will depend on the combined process metadata csv.
+    """
+
+    def requires(self):
+        # The meta files contain the path of the files in the data
+        # so we dont need to pass the extract as a dependency here.
+        return {
+            "meta": ConfigureProcessMetaData(outfile="process_metadata.csv"),
+        }
+
+
+class SubsamplePartitions(luigi_util.WorkTask):
+    """
+    Aggregates subsampling of all the partitions into a single task as dependencies.
+    All the subsampled files are stored in the requires workdir, so we just link to
+    that since there aren't any real outputs associated with this task.
+    This is a bit of a hack -- but it allows us to avoid rewriting
+    the Subsample task as well as take advantage of Luigi concurrency.
+    """
+
+    def requires(self):
+        # Perform subsampling on each partition independently
+        return {
+            "train": SubsamplePartition(
+                partition="train", max_files=config.MAX_TRAIN_FILES
+            ),
+            "test": SubsamplePartition(
+                partition="test", max_files=config.MAX_TEST_FILES
+            ),
+            "validation": SubsamplePartition(
+                partition="valid", max_files=config.MAX_VAL_FILES
+            ),
+        }
+
+    def run(self):
+        workdir = Path(self.workdir)
+        workdir.rmdir()
+        workdir.symlink_to(Path(self.requires()["train"].workdir).absolute())
+        self.mark_complete()
+
+
+class SplitTrainTestCorpus(luigi_util.SplitTrainTestCorpus):
```

Comment:
We can skip over the step that writes to mono and trims the length -- all nsynth samples are already all mono and exactly 64000 samples long (4s)

Reply:
Add this as a comment?

I guess I would still prefer to include this step, because people might be monkey-patching based upon this code to create new pipelines. And my preferences would just be that if its already mono and trimmed, that the pipeline step is quite fast.

Reply:
Okay -- if we want to still include it, but make it fast if it is already correct, then I we will need an additional step in the MonoTrimWav task that checks each sample first and bypasses the conversion / trim if it is already correct. I think there is an ffmpeg function - ffprobe?